main/bamboo/bamboo.go

```diff
@@ -171,6 +180,67 @@ func listenToZookeeper(conf configuration.Configuration, eventBus *event_bus.Eve
 	return serviceConn
 }
 
+func listenToEventStream(conf configuration.Configuration, eventBus *event_bus.EventBus) {
+	client := &http.Client{}
+	client.Timeout = 0 * time.Second
+
+	for _, marathon := range conf.Marathon.Endpoints() {
+		go func() {
+			for {
+				req, err := http.NewRequest("GET", marathon+"/v2/events", nil)
+				req.Header.Set("Accept", "text/event-stream")
+				if err != nil {
+					errorMsg := "An error occurred while creating request to Marathon events system: %s\n"
+					log.Printf(errorMsg, err)
+					continue
+				}
+
+				resp, err := client.Do(req)
+				if err != nil {
+					continue
```

Comment:
Sure, but always having instant reconnections is not what we want. You end up having Bamboo ddos your marathon master. So having a ticker that limits the maximum rate of reconnection to 1Hz is enough; if the connection stays up for over 1 second, the reconnection will be instant, if not, then it waits up to 1 second.